editor/src/components/editor/actions/actions.ts

```diff
@@ -4531,6 +4544,61 @@ export const UPDATE_FNS = {
       },
     }
   },
+  ADD_TAILWIND_CONFIG: (
+    action: AddTailwindConfig,
+    editor: EditorModel,
+    dispatch: EditorDispatch,
+  ): EditorModel => {
+    const packageJsonFile = getContentsTreeFileFromString(editor.projectContents, '/package.json')
+    const currentNpmDeps = dependenciesFromPackageJson(packageJsonFile, 'regular-only')
+    const updatedNpmDeps = [
+      ...currentNpmDeps,
+      requestedNpmDependency('postcss', PostCSSVersion),
+      requestedNpmDependency('tailwindcss', TailwindCSSVersion),
+    ]
+
+    fetchNodeModules(updatedNpmDeps, 'canvas').then((fetchNodeModulesResult) => {
+      const loadedPackagesStatus = createLoadedPackageStatusMapFromDependencies(
+        updatedNpmDeps,
+        fetchNodeModulesResult.dependenciesWithError,
+        fetchNodeModulesResult.dependenciesNotFound,
+      )
+      const packageErrorActions = Object.keys(loadedPackagesStatus).map((dependencyName) =>
+        setPackageStatus(dependencyName, loadedPackagesStatus[dependencyName].status),
+      )
+      dispatch([
+        ...packageErrorActions,
+        updateNodeModulesContents(fetchNodeModulesResult.nodeModules, 'full-build'),
+      ])
+    })
+
+    const projectContentsWithTailwind = addFileToProjectContents(
```

Comment:
I think for both of these we should maybe check if those files already exist first?

Reply:
fixed